src/lib/cost_estimation/cost_estimator_logical.cpp

```diff
@@ -14,66 +16,106 @@
 #include "types.hpp"
 #include "utils/assert.hpp"
 
+namespace {
+
+using namespace hyrise;  // NOLINT(build/namespaces)
+
+float expression_cost_multiplier(const std::shared_ptr<AbstractExpression>& expression) {
+  auto multiplier = 0.0f;
+
+  // Number of different columns accessed to factor in expression complexity. Also add a factor for correlated
+  // subqueries since we have to evaluate the subquery for each tuple again. In the past, we added to the factor for
+  // each expression in the predicate. This led to too pessimistic cost estimations for PredicateNodes compared to
+  // (semi-)joins. We start with a weight of 0 to ease, e.g., the estimation of column vs. column predicates of nested
```

Comment:
Can you briefly give an example of the old way (`column_a = 3 or column_b = 5`) loading to a large value so that is obvious that a join is cheaper in this case?

Further can you state what we have tried with correlated queries and that one should look into the PR to see the unsatisfactory results?

Reply:
Is this still open? Can I resolve?

Reply:
C'est ça?
https://github.com/hyrise/hyrise/blob/e905189b39185eac70a5d563ad6af3eb19d9c02f/src/lib/cost_estimation/cost_estimator_logical.cpp#L26-L38